--- a/include/download.mk
+++ b/include/download.mk
@@ -205,7 +205,10 @@ define DownloadMethod/rawgit
 	[ \! -d $(SUBDIR) ] && \
 	git clone $(OPTS) $(URL) $(SUBDIR) && \
 	(cd $(SUBDIR) && git checkout $(VERSION) && \
-	git submodule update --init --recursive) && \
+	git submodule update --init --recursive && \
+	echo "$(URL)" >.gitdownload_url && \
+	echo "$(VERSION)" >.gitdownload_version && \
+	git rev-parse HEAD >.gitdownload_rev) && \
 	echo "Packing checkout..." && \
 	export TAR_TIMESTAMP=`cd $(SUBDIR) && git log -1 --format='@%ct'` && \
 	rm -rf $(SUBDIR)/.git && \
