--- a/package/kernel/linux/modules/rtc.mk
+++ b/package/kernel/linux/modules/rtc.mk
@@ -220,6 +220,40 @@ endef
 
 $(eval $(call KernelPackage,rtc-rx8025))
 
+define KernelPackage/rtc-rv8803
+  SUBMENU:=$(RTC_MENU)
+  TITLE:=microcrystal RV-8803 / Epson RX-8803, RX-8900
+  DEFAULT:=m if ALL_KMODS && RTC_SUPPORT
+  DEPENDS:=+kmod-i2c-core
+  KCONFIG:=CONFIG_RTC_DRV_RV8803 \
+  CONFIG_RTC_CLASS=y
+  FILES:=$(LINUX_DIR)/drivers/rtc/rtc-rv8803.ko
+  AUTOLOAD:=$(call AutoLoad,50,rtc-rv8803,1)
+endef
+
+define KernelPackage/rtc-rv8803/description
+ Kernel module for microcrystal RV-8803 and Epson RX-8803/RX-8900 I2C RTC chips
+endef
+
+$(eval $(call KernelPackage,rtc-rv8803))
+
+define KernelPackage/rtc-rx8010
+  SUBMENU:=$(RTC_MENU)
+  TITLE:=Epson RX-8010SJ
+  DEFAULT:=m if ALL_KMODS && RTC_SUPPORT
+  DEPENDS:=+kmod-i2c-core +kmod-regmap-i2c
+  KCONFIG:=CONFIG_RTC_DRV_RX8010 \
+  CONFIG_RTC_CLASS=y
+  FILES:=$(LINUX_DIR)/drivers/rtc/rtc-rx8010.ko
+  AUTOLOAD:=$(call AutoLoad,50,rtc-rx8010,1)
+endef
+
+define KernelPackage/rtc-rx8010/description
+ Kernel module for Epson RX-8010SJ I2C RTC chip
+endef
+
+$(eval $(call KernelPackage,rtc-rx8010))
+
 define KernelPackage/rtc-s35390a
   SUBMENU:=$(RTC_MENU)
   TITLE:=Seico S-35390A
