--- a/config/Config-kernel.in
+++ b/config/Config-kernel.in
@@ -99,6 +99,23 @@ config KERNEL_RPI_AXIPERF
 	default y
 	depends on KERNEL_PERF_EVENTS && TARGET_bcm27xx
 
+config KERNEL_OF_OVERLAY
+	bool "Compile the kernel with device tree overlay support"
+	default n
+	select KERNEL_CONFIGFS_FS
+	help
+	  This option enables dynamic device tree fragment loading
+	  and undloading via CONFIGFS device-tree/overlays.
+
+config KERNEL_CONFIGFS_FS
+	bool "Compile the kernel with userspace-driven configuration filesystem"
+	default n
+	help
+	  configfs is a RAM-based filesystem that provides the converse
+	  of sysfs's functionality. Where sysfs is a filesystem-based
+	  view of kernel objects, configfs is a filesystem-based manager
+	  of kernel objects, or config_items.
+
 config KERNEL_UBSAN
 	bool "Compile the kernel with undefined behaviour sanity checker"
 	help
