--- a/config/Config-kernel.in
+++ b/config/Config-kernel.in
@@ -591,6 +591,42 @@ config KERNEL_BPF_KPROBE_OVERRIDE
 	depends on KERNEL_KPROBES
 	default n
 
+config KERNEL_GPIO_CDEV
+	bool "Compile the kernel with gpiod support (GPIO char device interface)"
+	default n
+	help
+	  Say Y here to add the character device /dev/gpiochipN interface
+	  for GPIOs. The character device allows userspace to control GPIOs
+	  using ioctl() operations.
+
+	  This is the modern way for userspace access and should/will
+	  replace /sys/class/gpio/... over time.
+
+config KERNEL_GPIO_CDEV_V1
+	bool "Compile the kernel with support for GPIO ABI Version 1"
+	default y
+	depends on KERNEL_GPIO_CDEV
+	help
+	  Say Y here to support version 1 of the GPIO CDEV ABI.
+
+	  This ABI version is deprecated.
+	  Please use the latest ABI for new developments.
+
+	  Needed for libgpiod 1.x (only - as of June 2022 - not-yet-released
+	  version 2.x will support v2 ABI)
+
+	  If unsure, say Y.
+
+config KERNEL_GPIO_SYSFS
+	bool "Compile the kernel with /sys/class/gpio/... GPIO support (sysfs interface)"
+	default y
+	help
+	  Say Y here to add the legacy sysfs interface for GPIOs.
+
+	  This ABI is deprecated. If you want to use GPIO from userspace,
+	  use the character device /dev/gpiochipN with the appropriate
+	  ioctl() operations instead.
+
 config KERNEL_AIO
 	bool "Compile the kernel with asynchronous IO support"
 	default y if !SMALL_FLASH
