--- a/feeds/packages/net/ola/Makefile
+++ b/feeds/packages/net/ola/Makefile
@@ -2,6 +2,7 @@
 # Copyright (C) 2006-2015 OpenWrt.org
 # Copyright (C) 2015-2017 Christian Beier <dontmind@freeshell.org>
 #
+# Patched to run as root 2018 by Lukas Zeller <luz@plan44.ch>
 # This is free software, licensed under the GNU General Public License v2.
 # See /LICENSE for more information.
 #
@@ -52,6 +53,7 @@ CONFIGURE_ARGS += \
 	--disable-fatal-warnings \
 	--disable-unittests \
 	--without-dns-sd \
+	--disable-root-check \
 	--with-ola-protoc-plugin="$(STAGING_DIR_HOSTPKG)/bin/ola_protoc_plugin"
 
 HOST_CONFIGURE_ARGS += \
--- a/feeds/packages/net/ola/files/olad.init
+++ b/feeds/packages/net/ola/files/olad.init
@@ -7,8 +7,9 @@ USE_PROCD=1
 START=90
 
 start_service() {
+	# e.g. when accessing Serial USB, olad needs the following dir for locks
+	mkdir -p /var/spool/uucp
 	procd_open_instance
-	procd_set_param user nobody
 	procd_set_param command /usr/bin/olad --syslog --log-level 2 --config-dir /etc/ola
 	procd_set_param respawn
 	procd_close_instance
