--- a/package/kernel/linux/modules/video.mk
+++ b/package/kernel/linux/modules/video.mk
@@ -267,6 +267,47 @@ endef
 $(eval $(call KernelPackage,fbtft-support))
 
 
+define KernelPackage/ssd1307fb
+  SUBMENU:=$(VIDEO_MENU)
+  TITLE:=SSD130x i2c OLED fb driver
+  DEPENDS:=\
+  kmod-i2c-core \
+  +kmod-fb \
+  +kmod-fb-sys-fops \
+  +kmod-fb-cfb-imgblt \
+  +kmod-fb-cfb-copyarea \
+  +kmod-fb-cfb-fillrect \
+  +kmod-backlight
+  KCONFIG:= \
+  CONFIG_FB \
+  CONFIG_FB_SSD1307 \
+  CONFIG_FB_BOTH_ENDIAN=y \
+  CONFIG_FB_CMDLINE=y \
+  CONFIG_FB_DEFERRED_IO=y \
+  CONFIG_FB_FOREIGN_ENDIAN=y \
+  CONFIG_FB_MODE_HELPERS=y \
+  CONFIG_FB_PROVIDE_GET_FB_UNMAPPED_AREA=n \
+  CONFIG_FB_TILEBLITTING=y \
+  CONFIG_LCD_CLASS_DEVICE \
+  CONFIG_PWM=y \
+  CONFIG_PWM_SYSFS=y
+  FILES:= \
+	$(LINUX_DIR)/drivers/video/fbdev/core/syscopyarea.ko \
+	$(LINUX_DIR)/drivers/video/fbdev/core/sysfillrect.ko \
+	$(LINUX_DIR)/drivers/video/fbdev/core/sysimgblt.ko \
+	$(LINUX_DIR)/drivers/video/fbdev/ssd1307fb.ko
+endef
+
+define KernelPackage/ssd1307fb/description
+ Solomon SSD130x small i2c OLED display fb driver (p44/luz)
+ Compatible with ssd1305, ssd1306, ssd1307 and ssd1309
+endef
+
+$(eval $(call KernelPackage,ssd1307fb))
+
+
+
+
 
 define KernelPackage/drm
   SUBMENU:=$(VIDEO_MENU)
