--- a/target/linux/ramips/dts/OMEGA2.dtsi
+++ b/target/linux/ramips/dts/OMEGA2.dtsi
@@ -16,20 +16,21 @@
 		system_led: system {
 			gpios = <&gpio1 12 GPIO_ACTIVE_LOW>;
 		};
-	};

-	gpio-keys-polled {
-		compatible = "gpio-keys-polled";
-		#address-cells = <1>;
-		#size-cells = <0>;
-		poll-interval = <20>;
+		status1 {
+			label = "p44:green:status1";
+			gpios = <&gpio1 4 GPIO_ACTIVE_HIGH>;
+			default-state = "on";
+		};

-		reset {
-			label = "reset";
-			gpios = <&gpio1 6 GPIO_ACTIVE_HIGH>;
-			linux,code = <KEY_RESTART>;
+		status2 {
+			label = "p44:red:status2";
+			gpios = <&gpio1 5 GPIO_ACTIVE_HIGH>;
+			default-state = "on";
 		};
+
 	};
+
 };

 &pinctrl {
