--- a/scripts/config/mconf-cfg.sh
+++ b/scripts/config/mconf-cfg.sh
@@ -33,6 +33,14 @@ if [ -f /usr/include/ncurses/ncurses.h ]
 	exit 0
 fi
 
+# Check apple silicon brew path
+if [ -f /opt/homebrew/opt/ncurses/include/ncurses.h ]; then
+	echo cflags=\"-D_GNU_SOURCE -I/opt/homebrew/opt/ncurses/include\"
+	echo libs=\"-lncurses\"
+	exit 0
+fi
+
+
 # As a final fallback before giving up, check if $HOSTCC knows of a default
 # ncurses installation (e.g. from a vendor-specific sysroot).
 if echo '#include <ncurses.h>' | ${HOSTCC} -E - >/dev/null 2>&1; then
