--- a/interface/vcos/pthreads/CMakeLists.txt
+++ b/interface/vcos/pthreads/CMakeLists.txt
@@ -1,7 +1,7 @@
 # MSVC5 does not fully support C99, enabling declaration-after-statement
 # warnings allows a common MSVC5 build error to be detected in Linux builds.
 if (CMAKE_COMPILER_IS_GNUCC)
-   set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Werror -Wdeclaration-after-statement")
+   set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Wdeclaration-after-statement")
    add_definitions (-D_GNU_SOURCE)
 endif ()
 
--- a/interface/vcos/CMakeLists.txt
+++ b/interface/vcos/CMakeLists.txt
@@ -43,7 +43,7 @@ foreach (header ${HEADERS})
 endforeach ()
 
 if (CMAKE_COMPILER_IS_GNUCC)
-   add_definitions (-ggdb -Werror -Wall)
+   add_definitions (-ggdb -Wall)
 endif ()
 
 if (CMAKE_COMPILER_2005)
--- a/interface/vmcs_host/linux/vcfiled/CMakeLists.txt
+++ b/interface/vmcs_host/linux/vcfiled/CMakeLists.txt
@@ -1,5 +1,5 @@
 
-add_definitions(-Werror)
+#add_definitions(-Werror)
 
 # vcfiled - serves files to videocore. used for media handlers from
 # OpenMAX/IL and loading VLLs.
@@ -14,7 +14,7 @@ target_link_libraries(vcfiled
                       vchiq_arm
                       vcos)
 
-install(TARGETS vcfiled 
+install(TARGETS vcfiled
         RUNTIME DESTINATION sbin)
 
 configure_file (etc/init.d/vcfiled ${PROJECT_BINARY_DIR}/etc/init.d/vcfiled)
--- a/interface/mmal/CMakeLists.txt
+++ b/interface/mmal/CMakeLists.txt
@@ -3,7 +3,7 @@ if (NOT DEFINED LIBRARY_TYPE)
 set(LIBRARY_TYPE SHARED)
 endif (NOT DEFINED LIBRARY_TYPE)
 
-add_definitions(-Wall -Werror)
+add_definitions(-Wall)
 
 add_library(mmal SHARED util/mmal_util.c)
 
--- a/host_applications/linux/libs/sm/CMakeLists.txt
+++ b/host_applications/linux/libs/sm/CMakeLists.txt
@@ -3,7 +3,7 @@ if (WIN32)
    set(VCOS_PLATFORM win32)
 else ()
    set(VCOS_PLATFORM pthreads)
-   add_definitions(-Wall -Werror)
+   add_definitions(-Wall)
 endif ()
 
 include_directories( ../../../..
--- a/host_applications/linux/libs/bcm_host/CMakeLists.txt
+++ b/host_applications/linux/libs/bcm_host/CMakeLists.txt
@@ -3,17 +3,17 @@ if (WIN32)
    set(VCOS_PLATFORM win32)
 else ()
    set(VCOS_PLATFORM pthreads)
-   add_definitions(-Wall -Werror)
+   add_definitions(-Wall)
 endif ()
 
-include_directories( ../../../.. 
+include_directories( ../../../..
 		     ../../../../interface/vcos/${VCOS_PLATFORM}
 		     ../../../../host_support/linux/added/include
 		     ../../../../host_applications/vmcs/test_apps/iltest
                      ../../../../host_applications/framework/common )
 
-add_library(bcm_host ${SHARED} bcm_host.c 
-       ../../../../interface/vmcs_host/linux/vcfilesys.c 
+add_library(bcm_host ${SHARED} bcm_host.c
+       ../../../../interface/vmcs_host/linux/vcfilesys.c
        ../../../../interface/vmcs_host/linux/vcfiled/vcfiled_check.c)
 
 target_link_libraries(bcm_host vcos vchostif)
--- a/host_applications/linux/apps/smem/CMakeLists.txt
+++ b/host_applications/linux/apps/smem/CMakeLists.txt
@@ -4,7 +4,7 @@ get_filename_component (VIDEOCORE_ROOT .
 include (${VIDEOCORE_ROOT}/makefiles/cmake/global_settings.cmake)
 
 if (NOT WIN32)
-   add_definitions(-Wall -Werror)
+   add_definitions(-Wall)
 endif ()
 
 include_directories (
--- a/host_applications/linux/apps/gencmd/CMakeLists.txt
+++ b/host_applications/linux/apps/gencmd/CMakeLists.txt
@@ -4,10 +4,10 @@ if (WIN32)
    set(VCOS_PLATFORM win32)
 else ()
    set(VCOS_PLATFORM pthreads)
-   add_definitions(-Wall -Werror)
+   add_definitions(-Wall)
 endif ()
 
-include_directories( ../../../.. 
+include_directories( ../../../..
                      ../../../../interface/vcos
                      ../../../../interface/vcos/${VCOS_PLATFORM} )
 
