--- a/feeds/packages/net/openssh/Makefile
+++ b/feeds/packages/net/openssh/Makefile
@@ -180,7 +180,13 @@ CONFIGURE_ARGS += \
 	--with-stackprotect \
 	--with$(if $(CONFIG_OPENSSL_ENGINE),,out)-ssl-engine \
 	--with$(if $(CONFIG_OPENSSH_LIBFIDO2),,out)-security-key-builtin
-	
+
+
+ifeq ($(ARCH),arm)
+CONFIGURE_ARGS += \
+	--with-cflags-after="-fzero-call-used-regs=skip"
+endif
+
 ifeq ($(BUILD_VARIANT),with-pam)
 CONFIGURE_ARGS += \
 	--with-pam
